src/layouts/settings/Profile.tsx

```diff
@@ -47,6 +48,17 @@ export function Profile() {
       .then(buf => new File([buf], filename, { type: mimeType }));
   };
 
+  const onPasswordValidate = (pass: string, valid: boolean) => {
+    var newUserDetails = { ...userSelfDetails };
+    if (pass && valid) {
+      newUserDetails.password = pass;
+    } else {
+      delete newUserDetails.password;
+    }
+
+    setUserSelfDetails({ ...newUserDetails });
+  };
+
   // open update dialog, when user was edited
   useEffect(() => {
     if (JSON.stringify(userSelfDetailsRedux) !== JSON.stringify(userSelfDetails)) {
```

Comment:
we have the lodash dependency. you can use [isEqual](https://lodash.com/docs/4.17.15#isEqual) here

Reply:
Legacy code, but I'll update it anyway